iroh/src/client.rs

```diff
@@ -624,6 +626,47 @@ where
         Ok(())
     }
 
+    /// Add an entry from a file path
+    pub async fn import_file(
+        &self,
+        author: AuthorId,
+        path: impl AsRef<Path>,
+        path_to_key: impl Fn(PathBuf) -> Bytes + Send + Sync + 'static,
```

Comment:
- are these bounds really needed, given that we immediately execute it?
- alternatively how about making the signature 

```rust
pub async fn import_file(
        &self,
        author: AuthorId,
        key: Bytes,
        path: impl AsRef<Path>,
)  -> Result<DocImportFileProgress> { }
```